packages/ai-jsx/src/react/index.ts

```diff
@@ -0,0 +1,84 @@
+import * as ReactModule from 'react';
+import * as LLMx from '../index.js';
+import { Serialize } from './serialize.js';
+export * from '../index.js';
+
+export declare namespace JSX {
+  type ElementType = ReactModule.JSX.ElementType | LLMx.JSX.ElementType | typeof React | typeof jsx;
+  type Element = ReactModule.JSX.Element & LLMx.Node;
+  type IntrinsicElements = ReactModule.JSX.IntrinsicElements;
+  type ElementChildrenAttribute = ReactModule.JSX.ElementChildrenAttribute & LLMx.JSX.ElementChildrenAttribute;
+}
+
+export function createElement(...args: Parameters<typeof ReactModule.createElement>) {
+  const tag = args[0];
+  const reactElement = ReactModule.createElement(...args);
+  const aiElement = LLMx.createElement(
+    tag === ReactModule.Fragment ? LLMx.Fragment : (tag as any),
+    args[1] as any,
+    ...args.slice(2)
+  );
+  return LLMx.makeIndirectNode(reactElement, aiElement);
+}
+
+export const Fragment = ReactModule.Fragment;
+
+export function useAI(children: LLMx.Node) {
+  const [result, setResult] = ReactModule.useState([] as ReactModule.ReactNode);
+  const [isDone, setIsDone] = ReactModule.useState(false);
+
+  ReactModule.useEffect(() => {
+    let shouldStop = false;
+    async function stream() {
+      setResult([]);
+      setIsDone(false);
+
+      // TODO: add a way for a render context to be aborted
+      const renderResult = LLMx.createRenderContext().render(children, {
+        stop: (e) => e.tag == React,
+        map: (frame) => frame,
+      });
+      for await (const frame of renderResult) {
+        if (shouldStop) {
+          return;
+        }
+
+        setResult(frame.map((e) => (LLMx.isElement(e) ? e.props.children : e)) as ReactModule.ReactNode);
```

Comment:
So we have an invariant that every element will either be a React node, or its direct children will be an array of React nodes? Or is there something else going on?

Reply:
Right, because we have
```
stop: (e) => e.tag == React,
```
the only elements will be `AI.React` elements whose `children` prop is treated as a `ReactNode`.